feature/account/edit/src/main/kotlin/app/k9mail/feature/account/edit/AccountEditExternalContract.kt

```diff
@@ -1,15 +1,24 @@
 package app.k9mail.feature.account.edit
 
-import app.k9mail.feature.account.common.domain.entity.Account
+import com.fsck.k9.mail.ServerSettings
 
 interface AccountEditExternalContract {
 
-    fun interface AccountUpdater {
-        suspend fun updateAccount(account: Account): AccountUpdaterResult
+    sealed interface AccountUpdaterResult {
+        data class Success(val message: String) : AccountUpdaterResult
+        data class Failure(val error: AccountUpdaterFailure) : AccountUpdaterResult
+    }
+
+    sealed interface AccountUpdaterFailure {
+        data class AccountNotFound(val accountUuid: String) : AccountUpdaterFailure
+        data class UnknownError(val error: Exception) : AccountUpdaterFailure
+    }
 
-        sealed interface AccountUpdaterResult {
-            data class Success(val accountId: String) : AccountUpdaterResult
-            data class Error(val message: String) : AccountUpdaterResult
-        }
+    fun interface AccountServerSettingsUpdater {
+        suspend fun updateServerSettings(
+            accountUuid: String,
+            isIncoming: Boolean,
+            serverSettings: ServerSettings,
+        ): AccountUpdaterResult
```

Comment:
Your right that this method could just return a value. Given our absence of a crash reporting system, unnoticed app failures could be a blind spot for us. We should think about a way to populate issues in a way that we could be actionable on in the future.

My idea was to wrap errors in a result and populate them up to the ui to an error view. This would enable us to give users the ability to send us error details by copying the message, creating a screenshot and maybe link to the issue tracker. But I learned that your not a big fan of results and this approach indeed has drawbacks, as it could be pretty invasive requireing every contributor to follow such a pattern.

This is not by any means set in stone and I'm happy to discuss further.